packages/material-ui/src/Slider/Slider.js

```diff
@@ -430,9 +430,23 @@ const Slider = React.forwardRef(function Slider(props, ref) {
     setOpen(-1);
   });
 
+  if (disabled && active !== -1) {
+    setActive(-1);
+  }
+  if (disabled && focusVisible !== -1) {
+    setFocusVisible(-1);
+  }
+
   const isRtl = theme.direction === 'rtl';
 
   const handleKeyDown = useEventCallback((event) => {
+    if (disabled) {
+      // This check is necessary because Firefox and Safari will keep focus
+      // on a disabled slider, even though we blur it:
```

Comment:
I think the new behavior is pretty confusing. We should either actually `blur()` it. Something like 

```js
useEnhancedEffect(() => {
  // doWeHaveARefToTheThumb?
  const { current: thumb } = whateverNameWeHaveGivenTheRef;
  if (disabled && thumb === document.activeElement) {
    thumb.blur()
  }
}, [disabled])
```

or not remove the focus-visible (and active?) stylings. 

Right now we remove the focus-visible style but in certain browsers it'll still be focused. That's confusing.

Reply:
The current behavior matches how our ButtonBase behaves and how a native button behaves (focus-ring gets removed, button stays focused). Unless there is references against this behavior in the web specifications, I think that we should respect the original design decisions of Firefox and Safari

Reply:
I understand the argument for consistency with ButtonBase. But this is incorrect behavior since it is not spec-compliant behavior and confirmed a bug in Firefox.

Reply:
> I think that we should respect the original design decisions of Firefox and Safari.

Then we shouldn't remove focus styling as browsers with these quirks don't do so as well.

Reply:
+1 for blurring https://github.com/mui-org/material-ui/pull/22247#discussion_r474223785

Reply:
Firefox does not remove the focus styling since the `:focus` pseudo class still applies. We just reset those styles and can't see it. A custom theme might still apply it.